internal/engine/compiler/engine.go

```diff
@@ -925,7 +929,7 @@ entry:
 			fn := calleeHostFunction.parent.goFunc
 			switch fn := fn.(type) {
 			case api.GoModuleFunction:
-				fn.Call(ce.ctx, callCtx.WithMemory(ce.memoryInstance), stack)
+				fn.Call(ce.ctx, ce.callerFunctionInstance.Module.CallCtx.WithMemory(ce.memoryInstance), stack)
```

Comment:
due to the chain here, feels like a comment is needed, maybe extracting as a variable.